testing/prow/gpu-operator.sh

```diff
@@ -194,10 +194,19 @@ EOF
     done
 }
 
-test_master_branch() {
+publish_master_bundle() {
     trap collect_must_gather EXIT
 
-    oc label ns/openshift-operators openshift.io/cluster-monitoring=true --overwrite
+    export CI_IMAGE_GPU_COMMIT_CI_IMAGE_UID="master"
+    deploy_commit "https://gitlab.com/nvidia/kubernetes/gpu-operator.git" "master"
+
+    prepare_cluster_for_gpu_operator
```

Comment:
Shouldn't this be before deploy_commit? Probably doesn't matter but looks more clear that way

Reply:
doesn't matter indeed,

I wrote it this say (and before that in `test_master_branch`) so that:.
1. the test fails early if the operator cannot be compiled, no need to fully prepare the cluster to crash when building the bundle
2. for `test_master_branch`, we deploy the GPU Operator with `0` GPU node, then scale it up to `1`. Just a slightly different test, just in case ...

best would be to have:
- build and deploy from bundle
- validate GPU Operator
- publish bundle

that wouldn't be hard even with the current code, just need to call first
```

    ./run_toolbox.py gpu_operator bundle_from_commit "${gpu_operator_git_repo}" \
                                             "${gpu_operator_git_ref}" \
                                             "${GPU_OPERATOR_QUAY_BUNDLE_PUSH_SECRET}" \
                                             "${GPU_OPERATOR_QUAY_BUNDLE_IMAGE_NAME}" \
                                             --tag_uid "${CI_IMAGE_GPU_COMMIT_CI_IMAGE_UID}" \
                                             --namespace "${OPERATOR_NAMESPACE}" \
                                             --with_validator=true \
                                             --publish_to_quay=false <=== mind this
```
then
```

    ./run_toolbox.py gpu_operator bundle_from_commit "${gpu_operator_git_repo}" \
                                             "${gpu_operator_git_ref}" \
                                             "${GPU_OPERATOR_QUAY_BUNDLE_PUSH_SECRET}" \
                                             "${GPU_OPERATOR_QUAY_BUNDLE_IMAGE_NAME}" \
                                             --tag_uid "${CI_IMAGE_GPU_COMMIT_CI_IMAGE_UID}" \
                                             --namespace "${OPERATOR_NAMESPACE}" \
                                             --with_validator=true \
                                             --publish_to_quay=true <=== mind that
```
but that will be future work ...